### DISCUSSION TITLE
Module: System Service Enumeration (AV, etc)

### DISCUSSION THREAD

**Discussion by aconite33** [blacklanternsecurity/bbot#728] [Ideas]
Add a module that will enumerate the running services on a host (E.g., EDR, AV, End Host Protection Suites)

Example Projects:

* https://github.com/SecureAuthCorp/impacket - Services.py
* https://github.com/Hackplayers/evil-winrm

**Comment by SpamFaux** [comment]
Waiting on #298 before development of this enhancement

**Comment by aconite33** [comment]
#298 has been merged

**Comment by aconite33** [comment]
Fixed in commit https://github.com/blacklanternsecurity/bbot-private/commit/13e18824463059de99d9b0c27fa1b9533bc9a9a6

**Comment by aconite33** [comment]
Bug with naabu scanning. Does not appropriately identify open ports.

Do we need to switch back to zmap?

**Comment by SpamFaux** [comment]
@aconite33 Is this worth bumping to high-priority?

**Comment by aconite33** [comment]
Yes, I would like to make this a priority before any other issues get worked.

**Comment by SpamFaux** [comment]
@TheTechromancer & @liquidsec FYI on the addition of the high-priority tag for this reopened issue.

**Comment by TheTechromancer** [comment]
Nmap module has been added in the `async-testing` branch:

https://github.com/blacklanternsecurity/bbot/commit/b4c785064146d767d7481553ea80b334f10a5018

**Comment by SpamFaux** [comment]
Last rounds of testing worked well with the nmap changes.

**Comment by SpamFaux** [comment]
Final piece is to review the CSV vs XLSX options.

**Reply by TheTechromancer** [reply]
Can you refresh my memory on this? I seem to recall there was a specific reason why we didn't like CSV as a format, but I can't remember exactly.

Personally if the CSV is working I would prefer we stick with it, since XLSX files are not easily readable/greppable from the CLI, and  integrating them will require adding a pip dependency.
